## Business Understanding

Data has been collected from different hospitals, community clinics, maternal health cares from the rural areas of Bangladesh through the IoT based risk monitoring system.

Age, Systolic Blood Pressure as SystolicBP, Diastolic BP as DiastolicBP, Blood Sugar as BS, Body Temperature as BodyTemp, HeartRate and RiskLevel. All these are the responsible and significant risk factors for maternal mortality, that is one of the main concern of SDG of UN.

source: Ahmed, M. (2020). Maternal Health Risk [Dataset]. UCI Machine Learning Repository. https://doi.org/10.24432/C5DP5D.

## DATA understanding 
- head and tail (printing out the first 5 or few entries and the last entries)
- shape (dimensions)
- columns
- datatypes
- info
- describe numerical (descriptive)
- describe categroical

In [26]:
import pandas as pd
import numpy as np
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from sklearn.datasets import load_iris

## load data set

In [27]:
maternal_df=pd.read_csv("data/Maternal Health Risk Data Set.csv",sep=",")

In [28]:
rows,columns=maternal_df.shape
output=f"This dataset has ROWS:{rows}|| features{columns}"
print(output)

This dataset has ROWS:1014|| features7


In [29]:
maternal_df.head(5)

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
0,25,130,80,15.0,98.0,86,high risk
1,35,140,90,13.0,98.0,70,high risk
2,29,90,70,8.0,100.0,80,high risk
3,30,140,85,7.0,98.0,70,high risk
4,35,120,60,6.1,98.0,76,low risk


In [30]:
maternal_df.tail(5)

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
1009,22,120,60,15.0,98.0,80,high risk
1010,55,120,90,18.0,98.0,60,high risk
1011,35,85,60,19.0,98.0,86,high risk
1012,43,120,90,18.0,98.0,70,high risk
1013,32,120,65,6.0,101.0,76,mid risk


In [31]:
maternal_df.dtypes

Age              int64
SystolicBP       int64
DiastolicBP      int64
BS             float64
BodyTemp       float64
HeartRate        int64
RiskLevel          str
dtype: object

In [38]:
maternal_df.isna().sum()

Age            0
SystolicBP     0
DiastolicBP    0
BS             0
BodyTemp       0
HeartRate      0
RiskLevel      0
dtype: int64

In [39]:
maternal_df1=maternal_df.dropna()

In [40]:
maternal_df1.describe()

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate
count,1014.000000,1014.000000,1014.000000,1014.000000,1014.000000,1014.000000
mean,29.871795,113.198225,76.460552,8.725986,98.665089,74.301775
std,13.474386,18.403913,13.885796,3.293532,1.371384,8.088702
min,10.000000,70.000000,49.000000,6.000000,98.000000,7.000000
25%,19.000000,100.000000,65.000000,6.900000,98.000000,70.000000
50%,26.000000,120.000000,80.000000,7.500000,98.000000,76.000000
75%,39.000000,120.000000,90.000000,8.000000,98.000000,80.000000
max,70.000000,160.000000,100.000000,19.000000,103.000000,90.000000


insight:

**age**
-the mean is approximately 29.9 years
- a Range of 10 to 70
- a sd od 13.474386 implying a Wide spread of age which includes very young and older mothers (both higher risk groups)
**blood pressure**
1. SystolicBP
-averagely at a Mean of approx 113
- a range of  70 → 160
2. DiastolicBP
- averagely Mean of  76
- a range0f  49 to 100
- conclusion-Some values are high (≥140/90) → possible hypertension risk

**heart rate**
- a low of 7 which is unrealistic== outlier and a highof 70
- averagely at 74.301775

In [44]:
maternal_df = maternal_df1[maternal_df1['HeartRate'] > 30]  # remove unrealistic values
maternal_df

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate,RiskLevel
0,25,130,80,15.0,98.0,86,high risk
1,35,140,90,13.0,98.0,70,high risk
2,29,90,70,8.0,100.0,80,high risk
3,30,140,85,7.0,98.0,70,high risk
4,35,120,60,6.1,98.0,76,low risk
...,...,...,...,...,...,...,...
1009,22,120,60,15.0,98.0,80,high risk
1010,55,120,90,18.0,98.0,60,high risk
1011,35,85,60,19.0,98.0,86,high risk
1012,43,120,90,18.0,98.0,70,high risk


In [45]:
maternal_df.describe()

,Age,SystolicBP,DiastolicBP,BS,BodyTemp,HeartRate
count,1012.000000,1012.000000,1012.000000,1012.000000,1012.000000,1012.000000
mean,29.899209,113.184783,76.463439,8.727619,98.666403,74.434783
std,13.473560,18.419618,13.899372,3.296583,1.372421,7.521857
min,10.000000,70.000000,49.000000,6.000000,98.000000,60.000000
25%,19.000000,100.000000,65.000000,6.900000,98.000000,70.000000
50%,26.000000,120.000000,80.000000,7.500000,98.000000,76.000000
75%,39.000000,120.000000,90.000000,8.000000,98.000000,80.000000
max,70.000000,160.000000,100.000000,19.000000,103.000000,90.000000


insight:
- Blood Sugar (BS) is the most influential factor on maternal RiskLevel.
- Blood Pressure (Systolic & Diastolic) and Age also significantly affect risk.Abnormal Blood Pressure or hypertension results to critical risk factor
- Body Temperature has minimal effect due to low variability.see a standard deviation of 1,
- Extreme Age a min of 10 years and a maxi of 70 years, very young or older mothers == higher pregnancy complications

In [47]:
maternal_df1.describe(include='object')

C:\Users\ADMIN\AppData\Local\Temp\ipykernel_23116\2011338303.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  maternal_df1.describe(include='object')


,RiskLevel
count,1014
unique,3
top,low risk
freq,406


In [34]:
maternal_df['RiskLevel'].value_counts()

RiskLevel
low risk     406
mid risk     336
high risk    272
Name: count, dtype: int64

insights:
- we have low maternal risk with a frequency of 406 

## hypothesis

- H₀ (Null Hypothesis):
There is no significant relationship between maternal health indicators (Age, SystolicBP, DiastolicBP, BS, BodyTemp, HeartRate) and the RiskLevel of maternal mortality.

- H₁ (Alternative Hypothesis):
At least one of these factors has a significant relationship with RiskLevel.

In [35]:
maternal_df['RiskLevel'].unique()

<StringArray>
['high risk', 'low risk', 'mid risk']
Length: 3, dtype: str

In [36]:
features = ['Age','SystolicBP','DiastolicBP','BS','BodyTemp','HeartRate']

for col in features:
    low = maternal_df[maternal_df['RiskLevel']=='low risk'][col]
    mid = maternal_df[maternal_df['RiskLevel']=='mid risk'][col]
    high = maternal_df[maternal_df['RiskLevel']=='high risk'][col]
    print(high)

    f_stat, p_value = stats.f_oneway(low, mid, high)

    print(f"{col}: F={f_stat:.4f}, p={p_value:.4f}")

0       25
1       35
2       29
3       30
5       23
        ..
1008    48
1009    22
1010    55
1011    35
1012    43
Name: Age, Length: 272, dtype: int64
Age: F=46.1140, p=0.0000
0       130
1       140
2        90
3       140
5       140
       ... 
1008    120
1009    120
1010    120
1011     85
1012    120
Name: SystolicBP, Length: 272, dtype: int64
SystolicBP: F=95.9030, p=0.0000
0       80
1       90
2       70
3       85
5       80
        ..
1008    80
1009    60
1010    90
1011    60
1012    90
Name: DiastolicBP, Length: 272, dtype: int64
DiastolicBP: F=84.9601, p=0.0000
0       15.00
1       13.00
2        8.00
3        7.00
5        7.01
        ...  
1008    11.00
1009    15.00
1010    18.00
1011    19.00
1012    18.00
Name: BS, Length: 272, dtype: float64
BS: F=331.1581, p=0.0000
0        98.0
1        98.0
2       100.0
3        98.0
5        98.0
        ...  
1008     98.0
1009     98.0
1010     98.0
1011     98.0
1012     98.0
Name: BodyTemp, Length: 272, dtype: flo

insight:
- p-value < 0.05 for all features hence Reject the null hypothesis (H₀) for every feature
- F-statistic measures the difference between risk groups relative to variation within groups:
- Age: F = 46.114 which is  very high hence strong differences between RiskLevel groups
- HeartRate: F = 20.4527which is moderate ,still significant but less than Age

In [37]:
##  SystolicBP vs RiskLevel
low = maternal_df[maternal_df['RiskLevel']=='low risk']['SystolicBP']
mid = maternal_df[maternal_df['RiskLevel']=='mid risk']['SystolicBP']
high = maternal_df[maternal_df['RiskLevel']=='high risk']['SystolicBP']

f_stat, p_value = stats.f_oneway(low, mid, high)

print(f"F-statistic: {f_stat}, p-value: {p_value}")

F-statistic: 95.90295152093445, p-value: 7.29219385948788e-39


insight:
- The F-statistic (95.9) is very large.The variation between RiskLevel groups is much greater than the variation within low, mid, high risks groups
- The groups (low, mid, high risk) have very different averages for the variable being tested

- p values is extremely small- 0.0000000000000000000000000000000000000729

- p-value < 0.05 so we Reject the Null Hypothesis (H₀)
- There is strong statistical evidence that at least one maternal health indicator (Age, Blood Pressure, Blood Sugar, Body Temperature, or Heart Rate) has a significant relationship with RiskLevel== not reject H1


in conclusion:
Reject the null hypothesis because There is a statistically significant relationship between the maternal health indicators and the RiskLevel of maternal mortality.